.github/workflows/build-backend.yml

```
@@ -0,0 +1,19 @@
+name: Build Backend
+
+on:
+  pull_request:
+    branches: [master]
+
+jobs:
+  build-backend:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "20.x"
+          cache: "npm"
+      - name: Install dependencies
+        run: cd backend && npm install
+      - name: Run build
+        run: cd backend && npm run build
```

vielleicht macht es so mehr sinn direkt mit caching und npm ci was intended ist für solche workflows?
könnte man auch für frontend anwenden

```suggestion
name: Build Backend

on:
  pull_request:
    branches: [master]

jobs:
  build-backend:
    runs-on: ubuntu-latest
    defaults:
      run:
        working-directory: backend
    steps:
      - uses: actions/checkout@v4

      - uses: actions/setup-node@v3
        with:
          node-version: "20.x"
          cache: "npm"

      - name: Cache node_modules
        id: cache-node-modules
        uses: actions/cache@v3
        with:
          path: backend/node_modules
          key: npm-${{ runner.os }}-${{ hashFiles('backend/package-lock.json') }}
          restore-keys: |
            npm-${{ runner.os }}-

      - name: Install dependencies
        if: steps.cache-node-modules.outputs.cache-hit != 'true'
        run: npm ci

      - name: Run build
        run: npm run build
```